### PR TITLE
Upgrade containerd to 1.5

### PR DESCRIPTION
**Pull request by xmudrii**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Upgrade containerd to 1.5, [since v1.5.11 is recommended for Kubernetes 1.24 clusters because of some CNI issues](https://kubernetes.io/blog/2022/05/03/kubernetes-1-24-release-announcement/#cni-version-related-breaking-change). Amazon Linux 2 is still using 1.4 because there's no containerd 1.5 available (see https://github.com/awslabs/amazon-eks-ami/issues/800 for more details).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
xref #2010

**Does this PR introduce a user-facing change?**:
```release-note
Upgrade containerd to 1.5. Amazon Linux 2 is still using containerd 1.4 because 1.5 is not available.
```

/assign @kron4eg

### CODE REVIEW

File: pkg/scripts/testdata/TestUpgradeKubeadmAndCNIAmazonLinux.golden [hunk 1]
```diff
@@ -74,7 +74,7 @@ sudo yum versionlock delete docker containerd || true
 
 sudo yum install -y \
 	docker-'19.03.*' \
-	containerd.io-'1.4.*'
+	containerd.io-'1.5.*'
```
Comment (kron4eg) [inline pkg/scripts/testdata/TestUpgradeKubeadmAndCNIAmazonLinux.golden:77]: Hmm, I think this should not happen

**Review by kron4eg — CHANGES_REQUESTED** [review]
Something is wrong with version on Amazon linux

File: pkg/scripts/testdata/TestUpgradeKubeadmAndCNIAmazonLinux.golden (hunk 1 above)
Comment (xmudrii) [reply pkg/scripts/testdata/TestUpgradeKubeadmAndCNIAmazonLinux.golden:77]: I found the issue, I'll try to push the fix later.

Comment (xmudrii) [reply pkg/scripts/testdata/TestUpgradeKubeadmAndCNIAmazonLinux.golden]: @kron4eg Fixed, PTAL.

**Pull request merged**